src/fat.rs

```diff
@@ -26,7 +26,7 @@ pub fn create_fat_filesystem(
         .truncate(true)
         .open(out_fat_path)
         .unwrap();
-    let fat_size_padded_and_rounded = ((needed_size + 1024 * 64 - 1) / MB + 1) * MB;
+    let fat_size_padded_and_rounded = ((needed_size + 1024 * 64 - 1) / MB + MB) * MB;
```

Comment:
Is that increasing the size by a terabyte? Note that the expression in the parentheses is already multiplied by `MB`, so now it ends up being `(... + MB) * MB`, which should be a TB if I'm not mistaken. Replacing the 1 by a 2 is probably a more storage efficient fix 😀

Reply:
Alright, doing something else then, hold on…

Reply:
Alright, I decided to revert it back to what it was before and instead simply append `+ MB` to the end of that long calculation, since adding a megabyte is far more reasonable than multiplying.

Reply:
Finally got this down to one commit but it took `git reset --hard <the first of my previous commits>` and `git pull https://github.com/rust-osdev/bootloader main --allow-unrelated-histories --rebase` to do it.